### ISSUE TITLE
Metric for robustness in real-world scenarios

### ISSUE THREAD

**Issue by alexriedel1** [openvinotoolkit/anomalib#199]
Hey all,
in real-world scenarios a useful metric for a classifier would be the anomaly-score-distance between all normal and anormal predictions in a test set. This would help to choose classifiers that are more confident in their predictions. 

Do you know if a metric exists that implements this in 2-class classification problems?

**Comment by alexriedel1** [comment]
Thats actually the mean (squared) error, also  referred to as brier score In this scenario 🙄

**Comment by samet-akcay** [comment]
@alexriedel1, do you think this is something we should be adding to the repo? If yes, would you be able to provide more detail regarding the use-case?

**Comment by samet-akcay** [comment]
@alexriedel1, thanks for clarifying this. This would indeed be useful in certain usecases. We could definitely add this metric to the repo. 

Our priority at the moment, however, is to make the metrics optional to the user. Metrics are currently hardcoded and computed for each run. Idealy, the user should be able to choose which metric they want to see. They could of course choose to see all of the metrics like what's happening now, but this is not so efficient in terms of memory use.

**Comment by alexriedel1** [comment]
> Our priority at the moment, however, is to make the metrics optional to the user. Metrics are currently hardcoded and computed for each run. Idealy, the user should be able to choose which metric they want to see. They could of course choose to see all of the metrics like what's happening now, but this is not so efficient in terms of memory use.

Should I write a PR addressing this?

**Comment by samet-akcay** [comment]
Thanks @alexriedel1. @djdameln is already working on a design for this. Meanwhile you could maybe add the Brier score implementation if that's alright with you?

**Comment by alexriedel1** [comment]
@samet-akcay @djdameln 
it would be great, if that implementation works in a way, where the torchmetric metrics can be defined in the config.yaml and are computed respectively. This would be an easy way to add custom metrics that are already part of torchmetrics (like MSE).

In kind of pseudecode it would look like:

`config.yaml`
```
model:
  metrics: 
    -F1
    -MeanSquaredError
```

`anomaly_module.py`
```
import torchmetrics

img_metrics = [getattr(torchmetrics, module)() for module in self.hparams.model.metrics]
self.image_metrics = MetricCollection(metrics, prefix="image_").cpu()
```

I hope this makes my intention a bit clear.

**Comment by samet-akcay** [comment]
@alexriedel1, thanks, it is clear. You could perhaps have a look at WiP PR #230.

**Comment by djdameln** [comment]
Solved with https://github.com/openvinotoolkit/anomalib/pull/230. The desired metric can be included by adding `MeanSquaredError` to the list of metrics.
